iceoryx_utils/include/iceoryx_utils/cxx/type_traits.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
```

Comment:
Adding type_traits and unique_ptr to utils should probably have been a separate PR. Easier/quicker to review/merge and it could be worked on in parallel.

Reply:
Yeah ... unfortunately they evolved simultaneously with the new API development :confounded:. Plus having separate PRs would delay the new API by at least a day or two.

Reply:
No worries about the development process, that's how things get developed. Are you sure about the 2 days extra? We can share the work packages. Maybe either @marthtz or @elBoberido can carve out the `unique_ptr` in parallel and we'll rebase this branch squashing the unneeded commits. I suppose that could be a good approach.